src/app/modules/item/components/chapter-children/chapter-children.component.ts

```diff
@@ -30,8 +54,22 @@ export class ChapterChildrenComponent implements OnChanges, OnDestroy {
         .get(this.itemData.item.id, this.itemData.currentResult.attemptId)
         .subscribe(
           children => {
-            this.children = children;
-            this.state = 'ready';
+            this.children = children.map(child => {
+              const res = bestAttemptFromResults(child.results);
+              return {
+                ...child,
+                isLocked: !canCurrentUserViewItemContent(child),
+                result: res === null ? undefined : {
+                  validated: res.validated,
+                  score: res.score,
+                },
+              };
+            });
+
+            if (this.children.length === 0) this.state = 'empty';
+            else if (this.itemData.currentResult?.validated ||
+              this.children.some(item => item.category === 'Validation' && item.result && !item.result.validated)) this.state = 'ready';
```

Comment:
I think the condition is still incorrect.

Reply:
fixed